### DISCUSSION TITLE
Ideas & UI design for the coming new features

### DISCUSSION THREAD

**Discussion by JunkFood02** [JunkFood02/Seal#107] [Ideas]
As the upstream library introduced the function of stopping a specified process, we're now able to implement many new features like canceling download tasks, concurrent downloading, and even download queue management. 
- https://github.com/yausername/youtubedl-android/pull/173 

Obviously, we shouldn't rush to roll out all these features. I'll be working on the feature of canceling download tasks, but haven't got any good ideas for the UI yet. Please feel free to share your thoughts in the comments section.

Edit: After the release of v1.0.5, the current focus of development is multitasking and download queue management. This could be a major update so please stay tuned for it

**Comment by JunkFood02** [comment]
Or maybe we can just imitate the way we stop a playlist download 🤔

**Comment by Marco-9456** [comment]
> As the upstream library introduced the function of stopping a specified process, we're now able to implement many new features like canceling download tasks, concurrent downloading, and even download queue management.
> 
> * [Allow process destroying by specifying string id yausername/youtubedl-android#173](https://github.com/yausername/youtubedl-android/pull/173)
> 
> Obviously, we shouldn't rush to roll out all these features. **`I'll be working on the feature of canceling download tasks, but haven't got any good ideas for the UI yet.`** Please feel free to share your thoughts in the comments section.

- About canceling the download operation UI.

Maybe you can make download button functionality change after hitting it so that it shows an ❌️ to cancel a running operation 🤔 
![Screenshot_20220809-193840_Seal](https://user-images.githubusercontent.com/41828058/183725050-4890497a-73e6-44e7-8a7f-c7f138f5479d.jpg)

**Reply by JunkFood02** [reply]
Maybe a bit weird IMO. Also, according to the [MD guidelines](https://m3.material.io/components/floating-action-button/guidelines), the FABs cannot be used to represent a "destructive" action.

**Reply by Marco-9456** [reply]
> Maybe a bit weird IMO. Also, according to the [MD guidelines](https://m3.material.io/components/floating-action-button/guidelines), the FABs cannot be used to represent a "destructive" action.

That sucks 😕
But i will think of something else...

**Comment by shuvashish76** [comment]


**Reply by svblsvljb** [reply]
Btw, you can use yt-dlp in termux for this.

**Comment by Donkey-Doug** [comment]


**Reply by JunkFood02** [reply]
Please file an separate issue for this

**Reply by Donkey-Doug** [reply]
Will do

**Reply by Donkey-Doug** [reply]
#134

**Comment by JunkFood02** [comment]
![gif-20220819-014613](https://user-images.githubusercontent.com/69683722/185466299-81cea5c7-9bd4-48d9-beb4-7d8ca5fe0d8b.gif)

**Reply by Marco-9456** [reply]
> ![gif-20220819-014613](https://user-images.githubusercontent.com/69683722/185466299-81cea5c7-9bd4-48d9-beb4-7d8ca5fe0d8b.gif)

Looks great, what about a pause - resume button 🤔

**Reply by JunkFood02** [reply]
> Looks great, what about a pause - resume button 🤔

yt-dlp wont let you "pause" a download I think. The icon and the behavior here could be modified dynamically, this button would have multiple operations due to the state of the task

**Reply by svblsvljb** [reply]
> yt-dlp wont let you "pause" a download I think.

But we can suspend (and resume) yt-dlp process like ctrl-z (^Z) in terminal.

**Reply by JunkFood02** [reply]
> But we can suspend (and resume) yt-dlp process like ctrl-z (^Z) in terminal.

We run python and yt-dlp in the **Android Environment**, without many basic components or commands (like `shell` or `systemctl`). So it would be hard to achieve this. Also, yt-dlp would leave the `.part` file in the target directory. You can resume a same download task with it.

**Reply by svblsvljb** [reply]
> Also, yt-dlp would leave the `.part` file in the target directory. You can resume a same download task with it.


Btw, yt-dlp resumes downloading from `.part` files by default (`--continue` flag is default)

**Reply by JunkFood02** [reply]
> > Also, yt-dlp would leave the `.part` file in the target directory. You can resume a same download task with it.
> 
> Btw, yt-dlp resumes from `.part` files by default (`--continue` flag is default)


Yes, so I think it's enough for resume a download after canceling it

**Comment by Spy-boy** [comment]
Can it have features of spotiFlyer so that we don't have to use two different applications with different UI?

**Reply by shuvashish76** [reply]
#48

**Comment by abdulrahmanjahari** [comment]


**Reply by abdulrahmanjahari** [reply]
Actually no problem, It's not a big thing. I love ur apps 👍 I'm looking for this kinda app for long time. Thank u appreciate it. Pls keep it simple 👍
